config/config_helpers_test.go

```diff
@@ -316,4 +317,86 @@ func TestResolveTerragruntConfigString(t *testing.T) {
 			assert.Equal(t, testCase.expectedOut, actualOut, "For string '%s' include %v and options %v", testCase.str, testCase.include, testCase.terragruntOptions)
 		}
 	}
+}
+
+func TestResolveEnvInterpolationConfigString(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		str			   string
+		include		   *IncludeConfig
+		terragruntOptions options.TerragruntOptions
+		expectedOut	   string
+		expectedErr	   error
+		env			   map[string]string
+	}{
+		{
+			"foo/${get_env()}/bar",
+			nil,
+			options.TerragruntOptions{TerragruntConfigPath: "/root/child/.terragrunt", NonInteractive: true},
+			"",
+			InvalidFunctionParameters(""),
+			nil,
+		},
+		{
+			"foo/${get_env(Invalid Parameters)}/bar",
+			nil,
+			options.TerragruntOptions{TerragruntConfigPath: "/root/child/.terragrunt", NonInteractive: true},
+			"",
+			InvalidFunctionParameters("Invalid Parameters"),
+			nil,
+		},
+		{
+			"foo/${get_env(ENV,\"\")}/bar",
+			nil,
+			options.TerragruntOptions{TerragruntConfigPath: "/root/child/.terragrunt", NonInteractive: true},
+			"",
+			InvalidFunctionParameters("ENV,\"\""),
+			nil,
+		},
+		{
+			"foo/${get_env(TEST_ENV_TERRAGRUNT_HIT,'')}/bar",
+			nil,
+			options.TerragruntOptions{TerragruntConfigPath: "/root/child/.terragrunt", NonInteractive: true},
+			"foo//bar",
+			nil,
+			map[string]string{"TEST_ENV_TERRAGRUNT_OTHER": "SOMETHING"},
+		},
+		{
+			"foo/${get_env(TEST_ENV_TERRAGRUNT_HIT,'DEFAULT')}/bar",
+			nil,
+			options.TerragruntOptions{TerragruntConfigPath: "/root/child/.terragrunt", NonInteractive: true},
+			"foo/DEFAULT/bar",
+			nil,
+			map[string]string{"TEST_ENV_TERRAGRUNT_OTHER": "SOMETHING"},
+		},
+		{
+			"foo/${get_env(TEST_ENV_TERRAGRUNT_HIT,'default')}/bar",
+			nil,
+			options.TerragruntOptions{TerragruntConfigPath: "/root/child/.terragrunt", NonInteractive: true},
+			"foo/HIT/bar",
+			nil,
+			map[string]string{"TEST_ENV_TERRAGRUNT_HIT": "HIT"},
+		},
+	}
+
+	for _, testCase := range testCases {
+		if len(testCase.env) > 0 {
+			for key, value := range testCase.env {
+				os.Setenv(key, value)
```

Comment:
This will not work if we have multiple tests running in parallel that use env vars...

Perhaps a way to resolve this is to add a field to `TerragruntOptions` called `Env`, which is a `map[string]string`. The default value for it, set by `parseTerragruntOptionsFromArgs`, will be the parsed version of `os.Environ()`. But at test time, you could provide a custom map of env vars.